File: buzu/src/test/java/br/net/buzu/sample/enums/Card.java
package br.net.buzu.sample.enums;

public class Card {

	private Suit suit;
	private int number;

	public Card() {

	}

	public Card(Suit suit, int number) {
		super();
		this.suit = suit;
		this.number = number;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + number;
		result = prime * result + ((suit == null) ? 0 : suit.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Card other = (Card) obj;
		if (number != other.number)
			return false;
		if (suit != other.suit)
			return false;
		return true;
	}

	@Override
	public String toString() {
		return "" + number + " of " + suit;
	}

	public Suit getSuit() {
		return suit;
	}

	public int getNumber() {
		return number;
	}

	public void setSuit(Suit suit) {
		this.suit = suit;
	}

	public void setNumber(int number) {
		this.number = number;
	}

}

File: ppl-lib/src/test/kotlin/br/net/buzu/sample/enums/Card.kt
package br.net.buzu.sample.enums

class Card {

    var suit: Suit? = null
    var number: Int = 0

    constructor()

    constructor(suit: Suit, number: Int) : super() {
        this.suit = suit
        this.number = number
    }

    override fun hashCode(): Int {
        val prime = 31
        var result = 1
        result = prime * result + number
        result = prime * result + if (suit == null) 0 else suit!!.hashCode()
        return result
    }

    override fun equals(obj: Any?): Boolean {
        if (this === obj)
            return true
        if (obj == null)
            return false
        if (javaClass != obj.javaClass)
            return false
        val other = obj as Card?
        return if (number != other!!.number) false else suit == other.suit
    }

    override fun toString(): String {
        return "$number domainOf $suit"
    }

}
